crates/ibc/src/core/ics03_connection/error.rs

```diff
@@ -166,5 +166,9 @@ define_error! {
                 reason: String,
             }
             | e | { format_args!("invalid client state: {0}", e.reason) },
+
+        Other
```

Comment:
What is an example use case for this error? Users are not expected to generate any IBC core errors. Same comment applies to the client error.

Reply:
Here's my specific code use case.
Here I'm deserializing the data to get other types of errors in the ibc-rs interface implementation layer, so I want to expose an item that can be used externally to pass any type of error.
```rust
let connections_path = ConnectionsPath(conn_id.clone()).to_string().as_bytes().to_vec();

		if <Connections<T>>::contains_key(&connections_path) {
			let data = <Connections<T>>::get(&connections_path);
			let ret = ConnectionEnd::decode_vec(&data)
				.map_err(|_| Ics03Error::implementation_specific())?;

			Ok(ret)
		} else {
			Err(Ics03Error::connection_mismatch(conn_id.clone()))
		}
```